kv/cachedb.go

```diff
@@ -0,0 +1,68 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+type (
+	cachedb struct {
+		mu        sync.RWMutex
+		memTables map[int64]*memdb
+	}
+
+	// MemManager add in executor and tikv for reduce query tikv
+	MemManager interface {
+		Set(tableID int64, key Key, values []byte) error
+		Get(ctx context.Context, tableID int64, key Key) ([]byte, error)
+		Release(tableID int64)
+	}
+)
+
+// Set set values in cache
+func (c *cachedb) Set(tableID int64, key Key, values []byte) error {
```

Comment:
This cache implementation seems to be tricky. This pattern only performs well for Point Get, and cannot work for other kind of operations. It may conflict to a feature that we have already planned: add read only table (not only Point Get is read only, but simply all operations over the table is read only and is cached). Can you improve it to be more general?

Reply:
/cc @SunRunAway

Reply:
We have design three goals to improve in next month's milestone, more general is one of them. We studied the competition system, show as below:

```
评分依据：
- 项目完成度，项目的分数。
- 性能提升比例，与原来分支的性能作对比。

建议：可以通过完成一些小的 PR 去拿分，分数也会是一个参考维度。

具体的得分以及项目评估还是评委来定的，他们会在所有项目提交完成之后来进行统一的评估和打分。
```

So we focus on improving point get performance. It's a better entrance for test our program. This cache is independent and we will optimization it for other kind of operations. But we will take some time to understand other operations. 
Will this affect your planning? If right, is there any other way to resolve conflicts?

Reply:
I think we could improve this in the future, as we already have `memTableReader`, it's just filling the cache with data that needs careful consideration. At the moment, there shouldn't be time to implement this part. Also, it seems that a team can't pick two programs? :)

Reply:
Yes, I agree that an improvement can be made in future to become a read only table. It doesn't need to be implemented in this PR. 

I'm only concerning about whether this improvement can be really made based on current code base / design. If we are going to implement a new feature, which includes the current one, but cannot utilize the current code base, then this PR might not good enough to be accepted into master.

Could you illustrate or briefly introduce how the current code base can be *improved* to implement a read only table? Or if it couldn't, could you alter the design or interface?

Reply:
We could use something like https://github.com/pingcap/tidb/pull/10673, which might be simpler to actually implement. Because although the `membuffer` is also used, the `membuffer` here is considered not to be updated. So, we can add read only cache logic to `iterTxnMemBuffer`, or simply implement three scan interfaces like pr above
![image](https://user-images.githubusercontent.com/4352397/98092729-f9004d00-1ec1-11eb-8378-cfe2fcf2acd9.png)

Reply:
Thanks for the supply information! I'm fine with it.